File: Python/BOJ/정렬/2910.py
import sys; readl = sys.stdin.readline

n, c = map(int, readl().split())
nums = list(map(int, readl().split()))

arr = []
dic = dict()

for num in nums:
  if num in dic:
    dic[num] += 1
  else:
    arr.append(num)
    dic[num] = 1

arr.sort(key = lambda x: dic[x], reverse=True)

for num in arr:
  for _ in range(dic[num]):
    print(num, end = " ")

File: Swift/BOJ/2000/2910.swift
//
//  2910.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/03.
//

import Foundation

let nc = readLine()!.split { $0 == " " }.map { Int($0)! }
let nums = readLine()!.split { $0 == " " }.map { Int($0)! }

var arr: [Int] = []
var dict: [Int: Int] = [:]

for num in nums {
    if let cnt = dict[num] {
        dict[num] = cnt + 1
    } else {
        arr.append(num)
        dict[num] = 1
    }
}

arr.sort { dict[$0]! > dict[$1]! }

for num in arr {
    for _ in 0..<dict[num]! {
        print("\(num)", terminator: " ")
    }
}
